apps/settings/src/store/app-api-store.ts

```diff
@@ -293,6 +295,15 @@ export const useAppApiStore = defineStore('app-api-apps', {
 				})
 		},
 
+		async fetchDockerDaemons() {
+			return axios.get(generateUrl('/apps/app_api/daemons'))
+				.then((res) => {
+					this.defaultDaemon = res.data.daemons.find((daemon: IDeployDaemon) => daemon.name === res.data.default_daemon_config)
+					this.dockerDaemons = res.data.daemons.filter((daemon: IDeployDaemon) => daemon.accepts_deploy_id === 'docker-install')
+					return res
+				})
```

Comment:
This is an async method, there is no need for `then`.
```suggestion
			const { data } = await axios.get(generateUrl('/apps/app_api/daemons'))
			this.defaultDaemon = data.daemons.find((daemon: IDeployDaemon) => daemon.name === res.data.default_daemon_config)
			this.dockerDaemons = data.daemons.filter((daemon: IDeployDaemon) => daemon.accepts_deploy_id === 'docker-install')
```

But you probably should have error handling

This is an async method, there is no need for `then`.
```suggestion
			try {
				const { data } = await axios.get(generateUrl('/apps/app_api/daemons'))
				this.defaultDaemon = data.daemons.find((daemon: IDeployDaemon) => daemon.name === res.data.default_daemon_config)
				this.dockerDaemons = data.daemons.filter((daemon: IDeployDaemon) => daemon.accepts_deploy_id === 'docker-install')
			} catch (error) {
				logger.error('[app-api] Failed to fetch daemons', { error })
				return false
			}
			return true
```